File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const answer = [];
    let curr = 1;

    for (let i = 0; i < nums.length; i++) {
        answer[i] = curr;
        curr *= nums[i];
    }    

    curr = 1;
    for (let i = nums.length - 1; i >= 0; i--) {
        answer[i] *= curr;
        curr *= nums[i];
    }

    return answer;    
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer = [0] * len(nums)
        curr = 1

        for i in range(len(nums)):
            answer[i] = curr
            curr *= nums[i]

        curr = 1
        for i in range(len(nums) - 1, -1, -1):
            answer[i] *= curr
            curr *= nums[i]

        return answer

            